src/HPolyhedron.jl

```diff
@@ -422,23 +425,29 @@ constraint system ``A x ≤ b`` to ``A \\text{inv}(M) y ≤ b``.
 """
 function linear_map(M::AbstractMatrix{N},
                     P::PT;
+                    check_invertibility::Bool=true,
                     cond_tol::Number=DEFAULT_COND_TOL
                    ) where {N<:Real, PT<:HPoly{N}}
-    if !isinvertible(M; cond_tol=cond_tol)
-        if P isa HPolyhedron
-            error("linear maps for polyhedra need to be invertible")
+    if check_invertibility
+        if !isinvertible(M; cond_tol=cond_tol)
+            if P isa HPolyhedron
+                error("linear maps for polyhedra need to be invertible")
+            end
+            # use the implementation for general polytopes
+            return invoke(linear_map, Tuple{typeof(M), AbstractPolytope{N}}, M, P)
```

Comment:
I do not like the flow of this method, in particular the duplication of the `invoke`.

Reply:
Suppose that we name it `use_vertex_representation`, and that it is set to `false`. Then it may happen that it tries to use the invertibility criterion but it fails. Hence the name of the flag is misleading in this situation.

But `check_invertibility` does not suffer from this because if it is `true` the check is made, and it computes without passing to v-rep if possible, and if `false` the check is not made and it always computes in the v-rep.

Reply:
> I do not like the flow of this method, in particular the duplication of the `invoke`.

I have uploaded a new version, which uses `invoke` only once. The idea is to use short-circuited OR for the cases that will make the function use the v-rep. I removed the check for P being a HPolyhedron and linear map being not invertible, because i think it is not needed. But let me note if there is some use case that i'm missing.

Reply:
> `use_vertex_representation`, and that it is set to `false`. Then it may happen that it tries to use the invertibility criterion

But it should not, that is exactly my point. If that option is used, we have to go to vertices. Otherwise, we may do whatever we want.

Reply:
> I have uploaded a new version, which uses `invoke` only once.

Yes, it is :+1: now, but read on :wink:

> I removed the check for P being a HPolyhedron and linear map being not invertible, because i think it is not needed.

It is needed because `vertices_list` is not working as this method would it expect to. I am worried that [this test](https://github.com/JuliaReach/LazySets.jl/blob/master/test/unit_Polyhedron.jl#L84) does not fail now.

Reply:
To be clear, i expect for an option `use_vertex_representation` that is set to `false` => it never uses the vertices. Otherwise we need to make it a kwarg: "always", "if_not_invertible".

Reply:
I would say:
* `use_vertex_representation == true` => use vertices
* `use_vertex_representation == false` => do whatever you want